### PR TITLE
output list-of items errors

### PR DESCRIPTION
**Pull request by etokareva**

### CODE REVIEW

File: demo/pages/validation/index.html [hunk 1]
```diff
@@ -34,5 +36,13 @@
 		             key="contact.address.state"
 		             length-equal="2">
 		</q-grid:rule>
+		<q-grid:rule for="column"
```
Comment (klumba12) [inline demo/pages/validation/index.html]: do we need to changed for="column" to for="cell"?

File: src/plugins/validation/validator.js [hunk 2]
```diff
@@ -21,7 +23,18 @@ class Validator extends Plugin {
 			};
 			const validData = this.validator.validate(target);
```
Comment (klumba12) [inline src/plugins/validation/validator.js]: isValid

File: src/plugins/validation/validator.js [hunk 3]
```diff
@@ -21,7 +23,18 @@ class Validator extends Plugin {
...
 			if (!validData) {
-				return this.validator.getErrors()[this.key];
+				const errors = this.validator.getErrors()[this.key];
+				if (isString(errors)) {
+					return errors;
+				} else if (isArray(errors) && isArray(this.value)) {
+					const result = errors.reduce((prev, curr, i) => {
```
Comment (klumba12) [inline src/plugins/validation/validator.js]: prev - memo
curr - error

File: src/plugins/validation/validator.js [hunk 4]
```diff
@@ -21,7 +23,18 @@ class Validator extends Plugin {
...
+						if (curr !== null) {
+							prev.push(`${this.value[i]}: ${curr}`);
+						}
+						return prev;
+					}, []);
+					return result.join('\n');
```
Comment (klumba12) [inline src/plugins/validation/validator.js]: can we always return array of strings? so UI template can decide how to show this?

File: demo/pages/validation/index.html (hunk 1 above)
Comment (etokareva) [reply demo/pages/validation/index.html]: I can change, OK

File: src/plugins/validation/validator.js (hunk 4 above)
Comment (etokareva) [reply src/plugins/validation/validator.js]: Yes, we can. it would be better

File: src/plugins/validation/validator.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
 import PluginComponent from '../plugin.component';
 import * as validationService from '@grid/core/validation/validation.service';
+import {isString} from '@grid/core/utility';
 
+const isArray = angular.isArray;
```
Comment (titovmx) [inline src/plugins/validation/validator.js]: Why don't you use ```isArray``` from utility?

Comment (etokareva) [reply src/plugins/validation/validator.js]: This is a good question about isArray. Firtly, I used Array.isArray ES6 method, but our interpreter cursed on me and showed me that I should use angular.isArray instead. So, I used angular.isArray and it is OK for now. But maybe in the future this code will be moved to the core. So this is reasonable to use utility isArray method there.

Comment (klumba12) [reply src/plugins/validation/validator.js]: @grid/core/utility contains isArray

Comment (etokareva) [reply src/plugins/validation/validator.js]: I know

**Review by klumba12 — APPROVED** [review]

**Pull request merged**